src/compiler/scale.ts

```diff
@@ -1,26 +1,26 @@
-'use strict';
-require('../globals');
-var util = require('../util'),
-  time = require('./time'),
-  colorbrewer = require('colorbrewer'),
-  interpolate = require('d3-color').interpolateHsl;
+/// <reference path="../colorbrewer.d.ts"/>
+/// <reference path="../d3-color.d.ts"/>
 
-var scale = module.exports = {};
+import 'colorbrewer';
+import {interpolateHsl} from 'd3-color';
 
-scale.names = function(props) {
+import * as util from '../util';
+import * as time from './time';
+
+export function names(props) {
   return util.keys(util.keys(props).reduce(function(a, x) {
     if (props[x] && props[x].scale) a[props[x].scale] = 1;
     return a;
   }, {}));
 };
 
-scale.defs = function(names, encoding, layout, stats, facet) {
+export function defs(names, encoding, layout, stats, facet) {
   return names.reduce(function(a, name) {
-    var scaleDef = {};
-
-    scaleDef.name = name;
-    var type = scaleDef.type = scale.type(name, encoding);
-    scaleDef.domain = scale.domain(encoding, name, type, facet);
+    var scaleDef = {
+      name: name,
+      type: type(name, encoding),
+      domain: domain(encoding, name, scaleDef.type, facet)
```

Comment:
Isn’t the diff here obviously different?  
On the left I declare `scaleDef` first, so it’s always defined.  
On the right, scaleDef is undefined.  

You also remove my `var type = scaleDef.type`